src/libsyncengine/update_detection/update_detector/node.cpp

```diff
@@ -187,4 +220,18 @@ SyncPath Node::getPath() const {
     return path;
 }
 
+bool Node::isParentValid(const std::shared_ptr<Node> parentNode) const {
+    for (auto &child : _childrenById) {
+        if (child.second == parentNode) {
+            return false;
+        }
+
+        if (!child.second->isParentValid(parentNode)) {
```

Comment:
If every parent node is set via `setParent` (advised), then the recursive call is superfluous.

Reply:
I did not find a way to avoid the recursion. However, I change the function to climb up the update tree instead of going down and checking every child.

Reply:
Changed in the latest commits: the method is now iterative.